azurerm/internal/services/desktopvirtualization/virtual_desktop_workspace_application_group_association_resource.go

```diff
@@ -0,0 +1,182 @@
+package desktopvirtualization
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/locks"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/desktopvirtualization/parse"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmVirtualDesktopWorkspaceApplicationGroupAssociation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmVirtualDesktopWorkspaceApplicationGroupAssociationCreate,
+		Read:   resourceArmVirtualDesktopWorkspaceApplicationGroupAssociationRead,
+		Delete: resourceArmVirtualDesktopWorkspaceApplicationGroupAssociationDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.DesktopVirtualizationWorkspaceID(id)
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"workspace_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"application_group_reference_id": {
```

Comment:
This might make more sense as
```suggestion
			"application_group_id": {
```

as i'm not sure the reference description makes much if any sense here?

Reply:
yeah I will change this to the suggestion and update the docs to reflect it.